Cython/Build/Dependencies.py

```diff
@@ -1040,6 +1044,34 @@ def copy_to_build_dir(filepath, root=os.getcwd()):
                     dir = os.path.dirname(c_file)
                     safe_makedirs_once(dir)
 
+                # write out the depfile, if requested
+                if depfile:
+                    dependencies = deps.all_dependencies(source)
+                    src_base_dir, _ = os.path.split(source)
+                    # paths below the base_dir are relative, otherwise absolute
+                    paths = []
+                    for fname in dependencies:
+                        if src_base_dir in fname:
+                            paths.append(os.path.relpath(fname, src_base_dir))
+                        else:
+                            paths.append(os.path.abspath(fname))
+
+                    # manually add missing *.pyx files for each *.pxd :
+                    # `cimport foo` means a possible dependence on a `foo.pyx`
+                    # but `all_dependencies(...)` above only list a `foo.pxd`
```

Comment:
Changes in a `.pyx` file should not trigger a rebuild of files that only depend on the `.pxd`. That's intentional. If anything changes in the binary interface, that should be reflected by the .pxd (or fail when building the other module itself).

Reply:
Not sure I follow. As a silly example (tests/build/depfile_package.srctree below) : `pkg/sub/test.pyx` does `from ..test cimport get_str`; the `get_str` function is declared in `pkg/test.pxd` and implemented in `pkg/test.pyx`. If the implementation changes (i.e. `pkg/test.pyx`), I'd naively expect a rebuild of the module which is built from `pkg/sub/test.pyx`, is not so?

Reply:
If the implementation changes, but the interface doesn't, then nothing that uses the interface needs to be rebuilt.

Reply:
Ah, so a full build system notices that `pkg/test.pyx` changes and marks its targets for a rebuild. Thanks, am rolling back this part of the patch. I *think* it's nearly done now, the last thing is to figure out what you meant by the prefix comparisons. Will push after that.